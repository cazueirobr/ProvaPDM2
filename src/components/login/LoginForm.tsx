import {Image, View, Text, TextInput, StyleSheet, Touchable, Button, TouchableOpacity } from 'react-native'
import React, { useState } from 'react'
import { router } from 'expo-router';

interface LoginFormProps {
  login: () => void
}




export default function LoginForm({login}:LoginFormProps) {
  const [usuario, setUsuario] = useState('');
  const [senha, setSenha] = useState('');


 // const login = () => {
  //  console.log("Executei a função");
   // if(usuario == "teste" && senha == "123"){
    //  router.push("/login/socata")
     // }
 // }
  return (
    <View style={styles.container}>
      <TextInput style={styles.input} placeholder='Usuário' onChangeText={newText => setUsuario(newText)} value={usuario}></TextInput>
      <TextInput secureTextEntry={true} style={styles.input} placeholder='Senha' onChangeText={newPassword => setSenha(newPassword)} value={senha}></TextInput>
      <TouchableOpacity onPress={login} testID='submitButton'>
        <Image style={styles.image} source={require("../../../assets/images/enviareditado.png")}></Image>

      </TouchableOpacity>
    </View>
  )

}
const styles = StyleSheet.create(
    {
        container:{
            flexDirection: "column",
            justifyContent: 'space-evenly',
            alignContent: "center",
        },
        input:{
            marginTop: 60,
            width: 250,
            borderWidth: 1,
            borderColor: 'gray',
            padding: 10

        },
          image: {
            marginTop: 60,
            width: 250,
            height: 50
          },

    }
  );