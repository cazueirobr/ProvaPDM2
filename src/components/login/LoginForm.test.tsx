import renderer from "react-test-renderer";
import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import LoginForm from "./LoginForm";


describe('Login form', () => {
  it("Renderiza o formulario", () => {
    const tree = renderer.create(<LoginForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("testa o botÃ£o de envio", () => {
    const loginMock = jest.fn();
    const { getByTestId } = render(<LoginForm login = {loginMock} />);

    console.log("Antes de clickar:", loginMock.mock.calls.length);
  
    const submitButton = getByTestId("submitButton");
    fireEvent.press(submitButton);

    console.log("Depois de clicar:", loginMock.mock.calls.length);
  
    expect(loginMock).toHaveBeenCalled();


  });

});

