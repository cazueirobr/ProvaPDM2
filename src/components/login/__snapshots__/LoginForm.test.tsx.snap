// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form Renderiza o formulario 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "flexDirection": "column",
      "justifyContent": "space-evenly",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="UsuÃ¡rio"
    style={
      {
        "borderColor": "gray",
        "borderWidth": 1,
        "marginTop": 60,
        "padding": 10,
        "width": 250,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Senha"
    secureTextEntry={true}
    style={
      {
        "borderColor": "gray",
        "borderWidth": 1,
        "marginTop": 60,
        "padding": 10,
        "width": 250,
      }
    }
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="submitButton"
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "marginTop": 60,
          "width": 250,
        }
      }
    />
  </View>
</View>
`;
